### DISCUSSION TITLE
A quick and easy way to make this a media player?

### DISCUSSION THREAD

**Discussion by droyholmes** [pgale/lineage_15.1_Installation_TSV#20] [Q&A]
Is there an easy way to get this to act as a target for music assistant? I'm still new to HAOS, but if I could get this to a point where my dad could sit at a table and pick songs for it to play from our digital library it would be an easy win for me.

Reading the long thread on the "My take on a home...." post on the HA forum someone does say they have it working, I'll work my way through and try to update with steps. Otherwise the post is pointless and admin can delete.

Sincerely sorry for unfocused premature post.

**Comment by pgale** [comment]
Two ways I've used to do this:

1) Running Fully Kiosk Browser on the TSV which exposes a music player that MA can use. 
2) (preferred) Installing snapcast on the TSV and using the built-in snapcast server in MA. 

Snapcast (snapdroid) can be installed from the Play store although there's a later version on the snapcast GitHub https://github.com/badaix/snapdroid

It's been running reliably on seven of my TSV's here. MA sees them and exposes the players to HA too. Some nice functions like TTS pre-announce and volume management. 

I'd recommend snapcast. The FKB method works too but isn't as fully featured.

**Reply by droyholmes** [reply]
I'll need to find an easy way for my dad to get from the music interface to the dashboard and back. Maybe one of the web GUIs can be embedded inside a dashboard. I haven't gotten to dashboards yet in my journey, just ZWave sensor proof of concept and then the display thing.

**Reply by pgale** [reply]
I've developed a set of dashboards (have a look at my dashboard Git) that use bubble card for popups. I use the 'horizontal button stack' at the bottom of the screen to popup panels containing web frames. One of those is MA. 

MA is a bit slow on the TSV but it is usable. Quite long initial loading times. Seems MA hasn't been very well tuned for slower devices.

**Reply by pgale** [reply]
![image](https://github.com/user-attachments/assets/e7863897-e474-47e3-8a78-b36e5670722d)

**Reply by droyholmes** [reply]
Do you have the audio coming out of the TSV there? I just noticed that it seems to think it is a Sonos Move.

**Reply by pgale** [reply]
I have Sonos devices too and tend to use them mostly. The TSV is capable but of course nowhere near as good as Sonos speakers.

**Reply by pgale** [reply]
So you know that the MA player can select any playback device? You can control what's playing on any connected player.

**Reply by droyholmes** [reply]
The Sonos speaker quality would definitely be better, my question was more related to the player provider. If you look at the [table](https://music-assistant.io/player-support/) about the different Player Providers I was wondering if you got the TVS to be recognized as a Sonos Player as the integration between MA and Sonos appears more fully featured.

I'm still getting my head around the system architecture. If I understand it correctly, I sort of see the system as having 4 components: Controller/Player/Server/Storage. However, a single device can simultaneously fill as many of those roles as needed. Does this sound about right?

In my case, I have the music library stored on my NAS device. I plan to install Music Assistant as an addon to HA on my HAOS-RPi. Then using MA's built in snapcast server, I want it to communicate with my TVS which will be acting as both a player and a controller using snapdroid. After that works properly, I have another RPi to setup as a headless snap player to demo multi-room sync playback.

Does that plan seem rational or am I missing something?

**Reply by droyholmes** [reply]
Upon further reading, I definitely need more reading. I would be interested in your take on my 4-component architecture understanding. Just to see how far off I am.
